src/MinimalDelegation.sol

```diff
@@ -151,26 +151,31 @@ contract MinimalDelegation is
 
         Settings settings = keySettings[keyHash];
         if (settings.isExpired()) revert IKeyManagement.KeyExpired();
+        uint40 expiry = settings.expiration();
 
         IHook hook = settings.hook();
         validationData = hook.hasPermission(HooksLib.VALIDATE_USER_OP_FLAG)
-            ? hook.validateUserOp(keyHash, userOp, userOpHash)
-            : _handleValidateUserOp(keyHash, signature, userOp, userOpHash);
+            ? hook.validateUserOp(keyHash, userOp, userOpHash, expiry)
```

Comment:
I would actually say don't pass the expiry here, if the hook wants to check expiry they can call the getter function on the main delegated contract